### PR TITLE
Fix crash & error at saving scene

### LINKED ISSUE #14948: Crash upon opening project
**Godot version:**
Godot 3.0 32d8b99

**OS/device including version:**
Linux Mint Cinnamon 18.3 Sylvia x64

**Issue description:**
I was working on a project when the editor suddenly crashed. I don't really remember what I did to cause it... the last thing I did was mess around with colours. Now the editor crashes whenever you try to open the project.

BACKTRACE:

```
user@ASUSZ170PG ~/Documents/Godot Projects/1GAW 2 $ gdb -e godot.x11.opt.tools.64
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
(gdb) r
Starting program: /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff7f2c700 (LWP 16374)]
No touch devices found
[New Thread 0x7fffee5df700 (LWP 16375)]
[New Thread 0x7fffea59d700 (LWP 16376)]
[New Thread 0x7fffe9d9c700 (LWP 16377)]
OpenGL ES 3.0 Renderer: GeForce GTX 980 Ti/PCIe/SSE2
GLES3: max ubo light: 409
GLES3: max ubo reflections: 455, ubo size: 144
[New Thread 0x7fffe814c700 (LWP 16378)]
ARVR: Registered interface: Native mobile

Thread 1 "godot.x11.opt.t" received signal SIGSEGV, Segmentation fault.
0x00000000010d30c3 in ?? ()
(gdb) c
Continuing.
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x354b0) [0x7ffff47cf4b0] (??:0)
[2] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64() [0x10d30c3] (/home/user/Repositories/godot/scene/main/node.cpp:343)
[3] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64() [0x16b5e8f] (/home/user/Repositories/godot/scene/resources/packed_scene.cpp:285)
[4] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64() [0x16b827d] (/home/user/Repositories/godot/scene/resources/packed_scene.cpp:1684)
[5] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64() [0x446cbc] (/home/user/Repositories/godot/main/main.cpp:1602)
[6] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64(main+0xa6) [0x41cc96] (/home/user/Repositories/godot/platform/x11/godot_x11.cpp:53)
[7] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0) [0x7ffff47ba830] (??:0)
[8] /home/user/Documents/Godot Projects/1GAW 2/godot.x11.opt.tools.64() [0x41ff69] (??:?)
-- END OF BACKTRACE --

Thread 1 "godot.x11.opt.t" received signal SIGABRT, Aborted.
0x00007ffff47cf428 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/unix/sysv/linux/raise.c:54
54	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) c
Continuing.
Couldn't get registers: No such process.
Couldn't get registers: No such process.
Couldn't get registers: No such process.
(gdb) [Thread 0x7fffe814c700 (LWP 16378) exited]
[Thread 0x7fffe9d9c700 (LWP 16377) exited]
[Thread 0x7fffea59d700 (LWP 16376) exited]
[Thread 0x7fffee5df700 (LWP 16375) exited]
[Thread 0x7ffff7f2c700 (LWP 16374) exited]

Program terminated with signal SIGABRT, Aborted.
The program no longer exists.
```

**Steps to reproduce:**
1. Download the sample project below
2. Try to run it

**Minimal reproduction project:**
[Sample project.zip](https://github.com/godotengine/godot/files/1583132/Sample.project.zip)

- [x] I searched the existing [GitHub issues](https://github.com/godotengine/godot/issues?utf8=%E2%9C%93&q=is%3Aissue+)
  for potential duplicates.

### PR DESCRIPTION
**Pull request by volzhs**
Fix #14948, #14975

### CODE REVIEW

**Comment by volzhs** [comment]
I couldn't figure out why the group is null in the first place.

**Comment by volzhs** [comment]
I'd like to dig it deeper. I will comment if I find something.

**Comment by volzhs** [comment]
I found it crashes when saving scene.
and it comes from instancing temporarily scene when saving.
so I fixed it another way.

**Comment by volzhs** [comment]
I guess #14975 issue is related this PR also, the temporarily instancing for saving.
probably there could be cleaner way and fix all this kind of problem.

**Comment by volzhs** [comment]
after I looked into #14975 issue, this PR seems a kind of hack.
there should be a better way to handle #14948 and #14975 at the same time.

**Comment by volzhs** [comment]
those #14948 and #14975 issue comes from here.


instancing scene from here [main.cpp](https://github.com/godotengine/godot/blob/master/main/main.cpp#L1602) not but added to viewport.

in #14948 case,
Camera2D is under KinematicBody2D, so Camera2D is added to "physics_process", but it's not in scene tree so it's group becomes `null` from here [Node::add_to_group](https://github.com/godotengine/godot/blob/master/scene/main/node.cpp#L1811-L1815)

so it crashes from here [Node::move_child](https://github.com/godotengine/godot/blob/master/scene/main/node.cpp#L343)

in #14975 case,
it's also instanciated from [main.cpp](https://github.com/godotengine/godot/blob/master/main/main.cpp#L1602) but it's not still added to viewport yet.
but properties is set in [SceneState::instance](https://github.com/godotengine/godot/blob/master/scene/resources/packed_scene.cpp#L254)
and then [Camera::get_camera_transform](https://github.com/godotengine/godot/blob/master/scene/3d/camera.cpp#L140-L143) called.
thus, `!in_inside_tree` error occurred from [Spatial::get_global_transform](https://github.com/godotengine/godot/blob/master/scene/3d/spatial.cpp#L265)

after all, i think it all comes from the same problem.
it's instanced but not added tree yet.

**Comment by volzhs** [comment]
I think @reduz can figure out. :)

**Comment by volzhs** [comment]
ok. I think I figured out what problem is.
in 2.1, `_set` is mostly used for instancing for properties.
so, in `_set`, value is set but not call `set_[property_name]` when instancing.

but 3.0, `set_[property_name]` is directly called.
so, it should take care of some cases.
I not sure but probably there might be more cases.

**Comment by Chaosus** [comment]
Yeah, its pretty annoying...  hope this will be merged soon..

File: scene/main/node.cpp [hunk 1]
```diff
@@ -340,7 +340,8 @@ void Node::move_child(Node *p_child, int p_pos) {
 		data.children[i]->notification(NOTIFICATION_MOVED_IN_PARENT);
 	}
 	for (const Map<StringName, GroupData>::Element *E = p_child->data.grouped.front(); E; E = E->next()) {
-		E->get().group->changed = true;
+		if (E->get().group)
```
Comment (reduz) [inline scene/main/node.cpp]: I'm surprised this issue wasn't found before..

**Pull request merged**